File: minimum_waiting_time/minimum_waiting_time.go
package main

import (
	"fmt"
	"sort"
)

func minimumWaitingTime(queries []int) int {
	sort.Ints(queries)
	totalWaitingTime := 0

	for i := 0; i < len(queries); i++ {
		queriesLeft := len(queries) - (i + 1)
		totalWaitingTime += queries[i] * queriesLeft
	}

	return totalWaitingTime
}

func main() {
	queries := []int{3, 2, 1, 2, 6}
	fmt.Println(minimumWaitingTime(queries))
}

File: minimum_waiting_time/minimum_waiting_time.ts
function minimumWaitingTime(queries: number[]): number {
    queries.sort((a, b) => a - b);
    let totalWaitingTime = 0;

    for (let i = 0; i < queries.length; i++) {
        const queriesLeft = queries.length - (i + 1);
        totalWaitingTime += queries[i] * queriesLeft;
    }

    return totalWaitingTime;
}

console.log(minimumWaitingTime([3, 2, 1, 2, 6]));
